libs/domain/cart/coupon/coupon.component.ts

```diff
@@ -0,0 +1,133 @@
+import { AuthService } from '@spryker-oryx/auth';
+import { CartComponentMixin, CartService } from '@spryker-oryx/cart';
+import { resolve } from '@spryker-oryx/di';
+import { ContentMixin } from '@spryker-oryx/experience';
+import { LocaleService } from '@spryker-oryx/i18n';
+import { NotificationService } from '@spryker-oryx/site';
+import { AlertType } from '@spryker-oryx/ui';
+import { ButtonSize, ButtonType } from '@spryker-oryx/ui/button';
+import { IconTypes } from '@spryker-oryx/ui/icon';
+import { ColorType } from '@spryker-oryx/ui/link';
+import { ClearIconAppearance } from '@spryker-oryx/ui/searchbox';
+import { hydrate, signal } from '@spryker-oryx/utilities';
+import { LitElement, TemplateResult, html } from 'lit';
+import { query, state } from 'lit/decorators.js';
+import { repeat } from 'lit/directives/repeat.js';
+import { couponStyles } from './coupon.styles';
+
+@hydrate({ event: ['window:load'] })
+export class CouponComponent extends CartComponentMixin(
+  ContentMixin(LitElement)
+) {
+  static styles = couponStyles;
+
+  @state() hasError = false;
+  @state() errorMessage: '' | string = '';
+
+  @query('input[name=coupon]') coupon?: HTMLInputElement;
+
+  protected authService = resolve(AuthService);
+  protected cartService = resolve(CartService);
+  protected notificationService = resolve(NotificationService);
+  protected localeService = resolve(LocaleService);
+
+  protected isAuthenticated = signal(this.authService.isAuthenticated());
+
+  private successMessage: '' | string = '';
+
+  protected override render(): TemplateResult | void {
+    if (this.$isEmpty() || !this.isAuthenticated()) {
+      return;
+    }
+
+    this.errorMessage =
+      this.coupon?.value === ''
+        ? `${this.i18n('coupon.insert-a-coupon')}`
+        : `${this.i18n('coupon.cart-code-can-not-be-added')}`;
+
+    this.successMessage = `${this.i18n('coupon.-successfully-applied')}`;
+
+    return html`
+      <oryx-layout>
+        <h3>${this.i18n('coupon.have-a-coupon')}?</h3>
+
+        <section>
+          <oryx-input
+            ?hasError="${this.hasError}"
+            .errorMessage="${this.hasError ? this.errorMessage : ''}"
+          >
+            <input
+              placeholder="${this.i18n('coupon.coupon-code')}"
+              name="coupon"
+            />
+            <oryx-icon
+              type=${IconTypes.Close}
+              class="clear-icon"
+              appearance=${ClearIconAppearance.Hover}
+              @click=${(): void => this.clear()}
+            ></oryx-icon>
```

Comment:
Did we add this in the design? Seems a bit buggy, as you'd expect it to not appear when the form control is empty. I think we do not need it.

Reply:
There is a DoD on the story regarding this one: 
`Clearing Coupon Codes: Users should have the ability to clear the entered coupon code for re-entry or removal.`
We can add option to not show if the input is empty

Reply:
I don't think it add's to much complex and maybe it's worth in terms of UX, let's have a quick sync on that.